x-pack/plugin/async-search/src/main/java/org/elasticsearch/xpack/search/MutableSearchResponse.java

```diff
@@ -213,4 +268,17 @@ private void failIfFrozen() {
         }
         return failures.toArray(ShardSearchFailure[]::new);
     }
+
+    private int getQueryFailuresCount() {
+        if (queryFailures == null) {
+            return 0;
+        }
+        int count = 0;
```

Comment:
nit:  you can do `return queryFailures.asList().size()` ?

Reply:
@jimczi Thanks for the suggestion. It seems that `asList()` method creates a additional list object. Instead, in 39dafa9, I've added `AtomicArray::nonNullLength` method.